schema/compose-spec.json

```diff
@@ -99,7 +99,27 @@
                 "target": {"type": "string"},
                 "shm_size": {"type": ["integer", "string"]},
                 "extra_hosts": {"$ref": "#/definitions/list_or_dict"},
-                "isolation": {"type": "string"}
+                "isolation": {"type": "string"},
+                "secrets": {
+                  "type": "array",
+                  "items": {
+                    "oneOf": [
+                      {"type": "string"},
+                      {
+                        "type": "object",
+                        "properties": {
+                          "source": {"type": "string"},
+                          "target": {"type": "string"},
+                          "uid": {"type": "string"},
+                          "gid": {"type": "string"},
+                          "mode": {"type": "number"}
+                        },
+                        "additionalProperties": false,
+                        "patternProperties": {"^x-": {}}
```

Comment:
this model is already declared twice for service configs and secrets, maybe this is a good opportunity to extract it into it's own definition and use a `$ref`

Reply:
Yes sure 👍